### PR TITLE
add gaussian sampling method with noise table

### PR DESCRIPTION
**Pull request by rical730**

### CODE REVIEW

File: deepes/src/cached_gaussian_sampling.cc [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+CachedGaussianSampling::~CachedGaussianSampling() {
+    delete[] _noise_cache;
+}
+
+void CachedGaussianSampling::load_config(const DeepESConfig& config) {
+    _std = config.gaussian_sampling().std();
+    set_seed(config.seed());
+    _cache_size = config.gaussian_sampling().cache_size();
+    _noise_cache = new float [_cache_size];
+	memset(_noise_cache, 0, _cache_size * sizeof(float));
```
Comment (TomorrowIsAnOtherDay) [inline deepes/src/cached_gaussian_sampling.cc]: Code alignment.

File: deepes/src/cached_gaussian_sampling.cc [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    for (int64_t i = 0; i < size; ++i) {
+        *(noise + i) = *(_noise_cache + index + i);
+    }
+    return index;
+}
+
+bool CachedGaussianSampling::resampling(int index, float* noise, int64_t size) {
+    if (_noise_cache == nullptr) {
+        LOG(ERROR) << "[ERROR] Please use load_config() first.";
+        return false;
```
Comment (TomorrowIsAnOtherDay) [inline deepes/src/cached_gaussian_sampling.cc]: Please define bool success at the beginning of the function.

File: deepes/src/cached_gaussian_sampling.cc [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    }
+    if (noise == nullptr) {
```
Comment (TomorrowIsAnOtherDay) [inline deepes/src/cached_gaussian_sampling.cc]: Please add logging for early return.

File: deepes/src/cached_gaussian_sampling.cc [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        return false;
+    }
+    if ((index > _cache_size - size) || (index < 0)) {
+        LOG(ERROR) << "[ERROR] Sampling index out of bound( 0 - " << _cache_size - size << " ).";
```
Comment (TomorrowIsAnOtherDay) [inline deepes/src/cached_gaussian_sampling.cc]:
```c++
success = false;
return success;
```

File: deepes/src/cached_gaussian_sampling.cc (hunk 1 above)
Comment (rical730) [reply deepes/src/cached_gaussian_sampling.cc]: done

File: deepes/src/cached_gaussian_sampling.cc (hunk 2 above)
Comment (rical730) [reply deepes/src/cached_gaussian_sampling.cc]: done

File: deepes/src/cached_gaussian_sampling.cc (hunk 3 above)
Comment (rical730) [reply deepes/src/cached_gaussian_sampling.cc]: done

File: deepes/src/cached_gaussian_sampling.cc (hunk 4 above)
Comment (rical730) [reply deepes/src/cached_gaussian_sampling.cc]: done

**Review by TomorrowIsAnOtherDay — APPROVED** [review]
looks good to me.

**Pull request merged**